### PR TITLE
feat: SQL preview modal for Query History

### PR DESCRIPTION
**Pull request by nytai**
### SUMMARY

Building off of https://github.com/apache/incubator-superset/pull/11574 this PR adds a preview/inspect modal to the Query History list view with a toggle for User inputted SQL and db executed SQL. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

before:
Screen Shot 2020-11-10 at 9 57 48 AM

After:
Screen Shot 2020-11-09 at 5 35 32 PM
Screen Shot 2020-11-09 at 5 35 39 PM

Screen Shot 2020-11-09 at 5 35 51 PM

Also, added copy query button to Saved Query preview modal
Screen Shot 2020-11-09 at 5 35 18 PM

### TEST PLAN

- unit tested
- manually tested:
  - empty User SQL
  - empty Executed SQL
  - previous button
  - next button 
  - open in sql lab button
  - copy query button
  - list view row highlight

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI (behind feature flag `SIP_34_QUERY_SEARCH_UI`)
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/CRUD/data/components/SyntaxHighlighterCopy/index.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    svg {
+      visibility: visible;
+    }
+  }
+  svg {
+    position: relative;
+    top: 40px;
+    left: 512px;
+    visibility: hidden;
+    margin: -4px;
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/components/SyntaxHighlighterCopy/index.tsx:42]: small nit, but do you need the negative margin if you have the top and left relative positions?

File: superset-frontend/src/views/CRUD/data/components/SyntaxHighlighterCopy/index.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        selection.removeAllRanges();
+      }
+      if (props.addSuccessToast) {
+        props.addSuccessToast(t('SQL Copied!'));
+      }
+    }
+  }
+  const syntaxHighlighterProps = { ...props };
+  delete syntaxHighlighterProps.addDangerToast;
+  delete syntaxHighlighterProps.addSuccessToast;
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/components/SyntaxHighlighterCopy/index.tsx:94]: would this work for you? 
`const { addDangerToast, addSuccessToast, ...syntaxHighlighterProps } = props;`

do you need to remove 'props.children' from the syntaxHighlighterProps as well?

File: superset-frontend/src/views/CRUD/data/components/SyntaxHighlighterCopy/index.tsx (hunk 1 above)
Comment (nytai) [reply superset-frontend/src/views/CRUD/data/components/SyntaxHighlighterCopy/index.tsx:42]: That negative margin gets rid of the space that would be occupied by the element if it were not relative positioned. If we were using absolute positioning we wouldn't need it.

File: superset-frontend/src/views/CRUD/data/hooks.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  const index = queries.findIndex(query => query.id === currentQueryId);
+  const [currentIndex, setCurrentIndex] = useState(index);
+  const [disablePrevious, setDisbalePrevious] = useState(false);
+  const [disableNext, setDisbaleNext] = useState(false);
+
+  function checkIndex() {
+    if (currentIndex === 0) {
+      setDisbalePrevious(true);
+    } else {
+      setDisbalePrevious(false);
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/hooks.ts:42]: could this be shortened to `checkDisablePrevious(currentIndex === 0)`

File: superset-frontend/src/views/CRUD/data/hooks.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    if (currentIndex === queries.length - 1) {
+      setDisbaleNext(true);
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/hooks.ts:46]: `setDisableNext(currentIndex === queries.length - 1)`?

File: superset-frontend/src/views/CRUD/data/components/SyntaxHighlighterCopy/index.tsx (hunk 2 above)
Comment (nytai) [reply superset-frontend/src/views/CRUD/data/components/SyntaxHighlighterCopy/index.tsx:94]: 👏 why didn't I think of that!

File: superset-frontend/src/views/CRUD/data/components/SyntaxHighlighterCopy/index.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import React from 'react';
+import { styled, t } from '@superset-ui/core';
+import { SyntaxHighlighterProps } from 'react-syntax-highlighter';
+import sql from 'react-syntax-highlighter/dist/cjs/languages/hljs/sql';
+import github from 'react-syntax-highlighter/dist/cjs/styles/hljs/github';
+import SyntaxHighlighter from 'react-syntax-highlighter/dist/cjs/light';
+import { ToastProps } from 'src/messageToasts/enhancers/withToasts';
+import Icon from 'src/components/Icon';
+
+SyntaxHighlighter.registerLanguage('sql', sql);
```
Comment (riahk) [inline superset-frontend/src/views/CRUD/data/components/SyntaxHighlighterCopy/index.tsx:28]: Would it make sense to add a language prop so we can use this component with `json`/`css`/etc.?

File: superset-frontend/src/views/CRUD/data/query/QueryList.tsx [hunk 6]
```diff
@@ -292,15 +289,18 @@ function QueryList({ addDangerToast, addSuccessToast }: QueryListProps) {
...
-            <StyledSyntaxHighlighter language="sql" style={github}>
-              {shortenSQL(sql)}
-            </StyledSyntaxHighlighter>
+            <div
+              tabIndex={0}
+              role="button"
+              data-test={`open-sql-preview-${id}`}
+              onClick={() => setQueryCurrentlyPreviewing(original)}
+            >
+              <StyledSyntaxHighlighter language="sql" style={github}>
```
Comment (riahk) [inline superset-frontend/src/views/CRUD/data/query/QueryList.tsx:300]: What's the difference between setting the `language` prop here and calling `SyntaxHighlighter.registerLanguage` above? Does the latter just load in the highlighting logic for while former explicitly sets it?

File: superset-frontend/src/views/CRUD/data/hooks.ts [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { useState, useEffect } from 'react';
+
+type BaseQueryObject = {
+  id: number;
+};
+export function useQueryPreviewState<D extends BaseQueryObject = any>({
```
Comment (riahk) [inline superset-frontend/src/views/CRUD/data/hooks.ts:24]: nice 👌

**Review by riahk — APPROVED** [review]
👍 just a couple questions!

File: superset-frontend/src/views/CRUD/data/query/QueryList.tsx (hunk 6 above)
Comment (nytai) [reply superset-frontend/src/views/CRUD/data/query/QueryList.tsx:300]: I think the SyntaxHighlighter ships with a few languages, and probably an interface to create your own. I think the `registerLanguage` step is to slim down the component and not include a bunch of unnecessary code that won't be used. I'll scan the codebase for other languages we use with SyntaxHighlighter and include those too.

File: superset-frontend/src/views/CRUD/data/components/SyntaxHighlighterCopy/index.tsx (hunk 5 above)
Comment (nytai) [reply superset-frontend/src/views/CRUD/data/components/SyntaxHighlighterCopy/index.tsx]: I added `'sql' | 'markdown' | 'html' | 'json'` which is all I could find in the code base. Adding more, as needed, should be pretty simple.

**Review by willbarrett — APPROVED** [review]

File: superset-frontend/jest.config.js [hunk 8]
```diff
@@ -26,7 +26,10 @@ module.exports = {
     '^spec/(.*)$': '<rootDir>/spec/$1',
   },
   testEnvironment: 'enzyme',
-  setupFilesAfterEnv: ['jest-enzyme', '<rootDir>/spec/helpers/shim.ts'],
+  setupFilesAfterEnv: [
+    '<rootDir>/node_modules/jest-enzyme/lib/index.js',
+    '<rootDir>/spec/helpers/shim.ts',
+  ],
```
Comment (ktmud) [inline superset-frontend/jest.config.js:32]: @nytai any reason for this change?

Comment (nytai) [reply superset-frontend/jest.config.js:32]: I was having issues with `shim.ts` not loading when I was running tests in vscode, I was digging through some docs and most of them were using absolute paths. Is it causing issues?

**Pull request merged**